src/iarc7_motion/iarc_tasks/track_roomba_task.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+
+import math
+import rospy
+import tf2_ros
+import tf2_geometry_msgs
+
+from geometry_msgs.msg import TwistStamped
+from geometry_msgs.msg import Point
+from geometry_msgs.msg import PointStamped
+
+
+from iarc7_msgs.msg import OdometryArray
+
+from .abstract_task import AbstractTask
+from iarc_tasks.task_states import (TaskRunning,
+                                    TaskDone,
+                                    TaskCanceled,
+                                    TaskAborted,
+                                    TaskFailed)
+from iarc_tasks.task_commands import (VelocityCommand,
+                                      ArmCommand,
+                                      NopCommand)
+
+from height_holder import HeightHolder
+
+class TrackObjectTaskState:
+    init = 0
+    track = 1
+    waiting = 2
+
+class TrackRoombaTask(AbstractTask):
+
+    def __init__(self, task_dictionary):
+
+        self._roomba_id = task_dictionary['frame_id']
+
+        self._roomba_id = self._roomba_id  + '/base_link'
+        
+        #p-terms
+        self._k_x = .35
+        self._k_y = .35
+
+        if self._roomba_id is None or len(self._roomba_id) < 2:
```

Comment:
Why are you checking the length of the roomba id? It could be any length as we have not specified a format for id's. Just that they will be presumably non-empty strings.